core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultMemoryHealthChecker.java

```diff
@@ -0,0 +1,51 @@
+package com.linecorp.armeria.server.healthcheck;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.lang.management.MemoryUsage;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.base.MoreObjects;
+
+public class DefaultMemoryHealthChecker implements HealthChecker {
+
+    private final double targetHeapMemoryUtilizationRate;
+    private final double targetNonHeapMemoryUtilizationRate;
+    private final double targetTotalMemoryUtilizationRate;
+
+    public DefaultMemoryHealthChecker(final double targetHeapMemoryUtilizationRate, final double targetNonHeapMemoryUtilizationRate, final double targetMemoryTotalUsage) {
+        this.targetHeapMemoryUtilizationRate = targetHeapMemoryUtilizationRate;
+        this.targetNonHeapMemoryUtilizationRate = targetNonHeapMemoryUtilizationRate;
+        this.targetTotalMemoryUtilizationRate = targetMemoryTotalUsage;
```

Comment:
Yeah, we could add some checks here, e.g.

```java
this.heapUsageRatioThreshold = validateRatio(heapUsageRatioThreshold, "heapUsageRatioThreshold");
this.nonHeapUsageRatioThreshold = validateRatio(nonHeapUsageRatioThreshold, "nonHeapUsageRatioThreshold");
this.totalUsageRatioThreshold = validateRatio(totalUsageRatioThreshold, "totalUsageRatioThreshold");
...

private static void validateRatio(double ratio, String paramName) {
    checkArgument(ratio >= 0 && ratio <= 1, "%s (expected: >= 0 and <= 1)", paramName);
}
```

Also, I suggested renaming the parameters and member fields as shown in the above example.

- Utilization -> Usage (because it's shorter)
- Memory -> '' (because it's already a part of the enclosing class name)
- Target -> Threshold (because target means something we aim for, but we don't really want that happen here)